### ISSUE TITLE
Add Riivolution support

### ISSUE THREAD

**Issue by H1dd3nM1nd** [JoshuaMKW/Codehandler-Wii#2]
I tried getting this to run in Riivolution, but it keeps giving me a green screen with both of the released bins. I edited the addresses to correctly match the new ones, but Riivolution wants none of it.

Here are the patches i made, mimicking the original Riiocarina ones (using the full codehandler)

````
<memory valuefile="codehandler.bin" offset="0x80001800" />
<memory offset="0x000018FA" value="8000" />
<memory offset="0x000018FE" value="2210" />
<memory valuefile="/codes/{$__gameid}{$__region}{$__maker}.gct" offset="0x80002210" />
````

**Comment by JoshuaMKW** [comment]
The codehandler _should_ be compatible, as it meets spec for all the riivolution requirements

Maybe try this?

<wiidisc version="1"> <options>
		<section name="Ocarina">
			<option name="Ocarina Handler">
				<choice name="VI Hook">
					<patch id="ocarinabase_vi" />
					<patch id="ocarinabase" />
				</choice>
				<choice name="KPAD Hook">
					<patch id="ocarinabase_kpad" />
					<patch id="ocarinabase" />
				</choice>
				<choice name="Joypad Hook">
					<patch id="ocarinabase_joypad" />
					<patch id="ocarinabase" />
				</choice>
				<choice name="GXDraw Hook">
					<patch id="ocarinabase_gxdraw" />
					<patch id="ocarinabase" />
				</choice>
				<choice name="GXFlush Hook">
					<patch id="ocarinabase_gxflush" />
					<patch id="ocarinabase" />
				</choice>
				<choice name="OSSleep Hook">
					<patch id="ocarinabase_ossleep" />
					<patch id="ocarinabase" />
				</choice>
				<choice name="AxNextFrame Hook">
					<patch id="ocarinabase_axnextframe" />
					<patch id="ocarinabase" />
				</choice>
			</option>
			<option name="ocarina_codes">
				<choice name="Enabled">
					<patch id="ocarinacodes" />
				</choice>
			</option>
		</section>
	</options>

	<patch id="ocarinabase_vi">
		<memory ocarina="true" value="7CE33B783887003438A7003838C7004C" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase_kpad">
		<memory ocarina="true" value="9A3F005E38AE0080389FFFFC7E0903A6" offset="0x800018A8" />
		
		<memory ocarina="true" value="801D0060901E0060801D0064901E0064801D0068901E0068" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase_joypad">
		<memory ocarina="true" value="3AB500013A73000C2C1500043B18000C" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase_gxdraw">
		<memory ocarina="true" value="3CA0CC01380000613C80450098058000" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase_gxflush">
		<memory ocarina="true" value="90010014800305FC2C00000041820008" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase_ossleep">
		<memory ocarina="true" value="90A402E0806502E4908502E42C030000" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase_axnextframe">
		<memory ocarina="true" value="3800000E7FE3FB78B005000038800080" offset="0x800018A8" />
	</patch>
	<patch id="ocarinabase">
		<memory valuefile="codehandler.bin" offset="0x80001800" />
	</patch>
	<patch id="ocarinacodes">
		<memory offset="0x000018FA" value="8000" />
		<memory offset="0x000018FE" value="2110" />
		<memory valuefile="/codes/{$__gameid}{$__region}{$__maker}.gct" offset="0x80002210" />
	</patch>
</wiidisc>

Note that the 2 memory offset writes really aren't needed, as the codehandler already has those offsets built in. You should be able to remove them safely. Let me know your results, trying each code hook, since certain games struggle (even crash) with the standard VI Hook when using riivolution

**Comment by H1dd3nM1nd** [comment]
I tried all hooks and none of them loaded the GCT, which at least didn't crash the game. I doubt it's the GCT itself, since it works with the older codehandler.

**Comment by JoshuaMKW** [comment]
Sadly, there's not much I can do for this considering I don't have a USB gecko I can debug a real wii with, and this codehandler works in other places. I imagine that riivolution has some hard coded checks for a codehandler, causing my version to fail even if it is functionally the same. I would just use the standard codehandler if you can't get this one to work. Soon though I will try loading my codehandler in riivolution, and I will update with the results

**Comment by H1dd3nM1nd** [comment]
No, Riivolution doesn't have any hardcoded checks for that. And i wanted to use this one specifically because i ran out of lines with the original.

**Comment by JoshuaMKW** [comment]
Alright. I won't be able to test today cause of work but I should have time tomorrow

**Comment by JoshuaMKW** [comment]
Sorry for the super long delay, but try opening the codehandler in a hex editor and changing the bytes near the end (0x60 0x00 0x00 0x00) to be (0x12 0x34 0x56 0x78)
